### PR TITLE
Measure source

### PR DESCRIPTION
**Pull request by okeefm**
This pull request provides a `MeasureSource` object that can pull measures out of MongoDB, either by `_id` (as a `string` or an `ObjectId`), or by `user_id` and `hqmf_set_id`. Also includes tests for the MeasureSource.

Pull requests into js-ecqm-engine require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

Merger: please **squash and merge**

**Submitter:** @okeefm
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-7
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @holmesie
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "jasmine": "^2.9.0"
   },
   "dependencies": {
-    "cqm-models": "projecttacoma/cqm-models",
+    "cqm-models": "projecttacoma/cqm-models#measure_model",
```
Comment (holmesie) [inline package.json]: I'm guessing the hash means a branch name? If so, should the measure model be merged before this, and this should just point to master? If my guess was incorrect, ignore this.

File: spec/models/measure_source_spec.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+describe('A MongoDB Measure Source', () => {
+  const connectionInfo = 'mongodb://127.0.0.1:27017/js-ecqme-test';
+  const measureSource = new MeasureSource(connectionInfo);
+  const mes = new Measure({
+    cms_id: 'CMS123',
+  });
+
+  describe('Finding a measure given an ID', () => {
+    beforeEach(() => {
+      spyOn(measureSource.CQLMeasure, 'findById').and.returnValue([null, mes]);
```
Comment (holmesie) [inline spec/models/measure_source_spec.js]: Nice!

File: lib/models/measure_source.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    if (typeof measureId === 'string') {
+      measureIdObj = mongoose.Types.ObjectId(measureId);
+    } else if (measureId instanceof mongoose.Types.ObjectId) {
+      measureIdObj = measureId;
+    }
+
+    if (measureIdObj === null) return TypeError('measureId must be string or ObjectId');
+
+    [err, this.measure] = this.CQLMeasure.findById(measureIdObj);
```
Comment (ljades) [inline lib/models/measure_source.js]: This (and in `findMeasureByUser`) is not correct syntax. You'll get back a query but it won't execute unless you pass a callback to findById, and in that object `this.measure` is set to the retrieved measure.

**Review by ljades — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -22,10 +22,12 @@
...
     "eslint-config-airbnb-base": "^12.1.0",
     "eslint-plugin-import": "^2.8.0",
-    "jasmine": "^2.9.0"
+    "jasmine": "^2.9.0",
+    "sinon-mongoose": "^2.1.1",
+    "sinon": "^4"
   },
   "dependencies": {
-    "cqm-models": "projecttacoma/cqm-models",
+    "cqm-models": "projecttacoma/cqm-models#measure_model",
```
Comment (ljades) [inline package.json]: Get rid of the branch.

File: package.json (hunk 1 above)
Comment (okeefm) [reply package.json]: That is correct, yes

**Review by ljades — CHANGES_REQUESTED** [review]

**Review by ljades — APPROVED** [review]

**Review by holmesie — APPROVED** [review]
My comments were addressed; and questions answered. Looks good.
